pkg/distributor/distributor.go

```diff
@@ -2094,17 +2123,23 @@ func (d *Distributor) ActiveSeries(ctx context.Context, matchers []*labels.Match
 	if err != nil {
 		return nil, err
 	}
+	if nativeHistograms {
+		req.Type = ingester_client.NATIVE_HISTOGRAM_SERIES
+	}
 
 	tenantID, err := tenant.TenantID(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	maxResponseSize := math.MaxInt
-	if limit := d.limits.ActiveSeriesResultsMaxSizeBytes(tenantID); limit > 0 {
-		maxResponseSize = limit
+	// For native histograms we are going to return metrics level information so we don't need to limit the response size.
+	if !nativeHistograms {
+		if limit := d.limits.ActiveSeriesResultsMaxSizeBytes(tenantID); limit > 0 {
+			maxResponseSize = limit
+		}
```

Comment:
Yeah, I'm not sure about this one. My problem is the name of the limit , it's called `querier.active-series-results-max-size-bytes` , but we'd be limiting the intermediate result size.  Something more relevant could be `querier.max-fetched-series-per-query`, but that's for queries which have very different semantics than active series requests.

I'll add restore it and update the help text to try and make it clear how it's used.

Reply:
Done, CI running now

Reply:
From a user perspective, I'm not sure if the new help text is clearer. What exactly is your issue with
`Maximum size of an active series request result shard in bytes`? 

> but we'd be limiting the intermediate result size

that's exactly what we want though to avoid high in-use space in queriers if the shard count is too low

Reply:
> From a user perspective, I'm not sure if the new help text is clearer. What exactly is your issue with
Maximum size of an active series request result shard in bytes?
I think I just missed the word "shard" in it. I'll update to just add active native histogram metrics to the existing sentence.